finetuner/tuner/augmentation.py

```diff
@@ -16,22 +17,61 @@ def vision_preprocessor(
     :param width: image width.
     :param default_channel_axis: The color channel of the input image, by default -1, the expected input is H, W, C.
     :param target_channel_axis: The color channel of the output image, by default 0, the expected output is C, H, W.
+    :param phrase: phrase of experiment, either `train` or `validation`. At `validation` phrase, will not apply
+      random transformation.
     """
 
     def preprocess_fn(doc):
         return _vision_preprocessor(
-            doc, height, width, default_channel_axis, target_channel_axis
+            doc, height, width, default_channel_axis, target_channel_axis, phrase
         )
 
     return preprocess_fn
 
 
+def _set_image_blob_normalization(
+    blob,
+    channel_axis: int = -1,
+    img_mean=(0.485, 0.456, 0.406),
+    img_std=(0.229, 0.224, 0.225),
+) -> np.ndarray:
+    """Normalize a uint8 image :attr:`.blob` into a float32 image :attr:`.blob` inplace.
+
+    Following Pytorch standard, the image must be in the shape of shape (3 x H x W) and
+    will be normalized in to a range of [0, 1] and then
+    normalized using mean = [0.485, 0.456, 0.406] and std = [0.229, 0.224, 0.225]. These two arrays are computed
+    based on millions of images. If you want to train from scratch on your own dataset, you can calculate the new
+    mean and std. Otherwise, using the Imagenet pretrianed model with its own mean and std is recommended.
+
+    :param channel_axis: the axis id of the color channel, ``-1`` indicates the color channel info at the last axis
+    :param img_mean: the mean of all images
+    :param img_std: the standard deviation of all images
+    :return: itself after processed
+
+    .. warning::
+        Please do NOT generalize this function to gray scale, black/white image, it does not make any sense for
+        non RGB image. if you look at their MNIST examples, the mean and stddev are 1-dimensional
+        (since the inputs are greyscale-- no RGB channels).
+
+
+    """
+    blob = (blob / 255.0).astype(np.float32)
+    blob = np.moveaxis(blob, channel_axis, 0)
+    mean = np.asarray(img_mean, dtype=np.float32)
+    std = np.asarray(img_std, dtype=np.float32)
+    blob = (blob - mean[:, None, None]) / std[:, None, None]
```

Comment:
Imho, the normalization should be done with a pixel value mean that is computed of the whole dataset. 
If one image is super dark and another one is very bright, both will end up to show average lighting conditions, when normalizing per-image.

Reply:
could be, but it's tricky to get global mean value in a dataset. This piece of code is actually copied from jina core (it's in the image mixin). @gvondulong

Reply:
Yeah this is the standard imagenet normalization - it's used in nearly all CV applications (unless your domain is something completely different, like tissue cells)